modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/classloader/AbstractArtifactClassLoaderConfigurationAssembler.java

```diff
@@ -98,15 +118,75 @@ private List<URL> loadUrls(File artifactFile,
                              ClassLoaderModelBuilder classLoaderConfigurationBuilder) {
     final List<URL> dependenciesArtifactsUrls = new ArrayList<>();
 
-    // TODO W-11202141 - consider artifact patches for the case this is run within a Runtime
-
     dependenciesArtifactsUrls.add(getUrl(artifactFile));
     dependenciesArtifactsUrls.addAll(addArtifactSpecificClassLoaderConfiguration(classLoaderConfigurationBuilder));
-    dependenciesArtifactsUrls.addAll(addDependenciesToClasspathUrls(artifactFile, dependencies));
+    dependenciesArtifactsUrls.addAll(addDependenciesToClasspathUrls(dependencies));
 
     return dependenciesArtifactsUrls;
   }
 
+  private List<URL> getArtifactPatches() {
```

Comment:
i think this would be better passed from the client code rather than calculated here